QamarDeen/QamarDeen/Services/NotificationScheduler.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  NotificationScheduler.swift
+//  QamarDeen
+//
+//  Created by Mazyad Alabduljaleel on 1/19/17.
+//  Copyright © 2017 Batoul Apps. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+/// responsible for wrapping the Local Notification APIs with
+/// convenient access to manage the required reminders
+final class NotificationScheduler {
+    
+    class var canScheduleAlerts: Bool {
+        
+        let app = UIApplication.shared
+        if let settings = app.currentUserNotificationSettings , settings.types != UIUserNotificationType() {
+            return true // settings enabled
+        }
+        
+        let types: UIUserNotificationType = [.sound, .alert]
+        let notifSettings = UIUserNotificationSettings(types: types, categories: nil)
+        
+        app.registerUserNotificationSettings(notifSettings)
```

Comment:
Great point, that would definitely be the optimal choice. I've not added yet, since I think this is the responsibility of the caller (settings class).

The notification service would return `false`, and the settings class would wait for the permissions to be granted before trying to enable the setting again. This would keep the settings class stateful, which it will definitely be, and the NotificationScheduler stateless and simple.